### ISSUE TITLE
When generating error statuses, Content-Location may be incorrect

### ISSUE THREAD

**Issue by nevali** [bbcarchdev/spindle#66]
For example, a 404 for an item:

```
$ /opt/res/bin/quilt /734b9e3c-ba7f-4f4c-b6ed-635269e92691
quilt: Error: Status 404: No resource matching the request could be found.
Status: 404 Not found
Content-Type: text/turtle; charset=utf-8
Content-Location: /.ttl
Vary: Accept
Server: Quilt/0.0~160308.3340870.dirty

```

This is because the UUID is not consumed before returning the 404.

Internal tracking: [RESDATA-1177](https://jira.dev.bbc.co.uk/browse/RESDATA-1177)

**Comment by prasdeva** [comment]
Parking this ticket with below observations

Above issue is only seen when running quilt when deployed using docker-compose at the root level. This particular issue is not seen with quilt when built/run [building part of the tree](https://github.com/bbcarchdev/acropolis#re-building-part-of-the-tree)

Reason being, while latter uses root/quilt/quilt.conf where engines 'file' and 'resourcegraphs' are loaded while the farmer approach refer to root/docker/quilt.conf with just 'spindle' as engine. Spindle module is not setting the required canonical path(null) which results in content location of */.ttl* during error case.

Another minor thing. While browsing the code with in root/quilt/libquitl/canon.c::quilt_canon_str, I could see that length of memory buffer is incremented by 2 bytes (un intentionally ?) while reading the flag QCO_NOPARAMS for params if I am not missing anything. For the example with in description, total memory allocated is 8 bytes while required is 6 bytes.

**Comment by nevali** [comment]
The effort of determining whether those 2 extra bytes should be allocated outweighs the cost of just allocating them blindly, they're not lost -- the buffer's just allocated to be slightly larger than the string it contains, which is no big deal.

**Comment by nevali** [comment]
Resolved via bbcarchdev/quilt@39bad5fc3eaea2b8442cbd0021a27fb714bf9410
